packages/main/src/services/discord/index.ts

```diff
@@ -10,29 +9,56 @@ import Logger, { $mainLogger } from '../logger';
 class Discord {
   private rpc: DiscordRPC.Client;
   private isReady = false;
+  private baseStart: number;
+  private pauseStart: number;
+  private pausedTotal = 0;
+  private activity: {
+    details: string;
+    startTimestamp: number;
+    largeImageKey: string;
+  };
 
   constructor(
     @inject(Config) private config: Config,
     @inject($mainLogger) private logger: Logger
   ) {}
 
-  private async sendActivity(track: NuclearMeta) {
+  private async sendActivity() {
     try {
-      await this.rpc.setActivity({
-        details: `${track.artist} - ${track.name}`,
-        startTimestamp: Date.now(),
-        largeImageKey: 'logo'
-      });
+      await this.rpc.setActivity(this.activity);
+      this.logger.log('update discord activity');
     } catch (err) {
       this.logger.error('error trying to set discord activity');
     }
   }
 
-  async init(cb?: () => any) {
+  async pause() {
+    if (this.isReady && this.activity) {
+      this.pauseStart = Date.now();
+  
+      this.activity.details += '\nPaused';
+      this.activity.startTimestamp = this.pauseStart;
+      return this.sendActivity();
+    }
+  }
+
+  async play() {
+    if (this.isReady && this.activity) {
+      this.pausedTotal += Date.now() - this.pauseStart;
+      if (this.activity) {
+        this.activity.details = this.activity.details.substr(0, this.activity.details.length - 8);
```

Comment:
```suggestion
        this.activity.details = this.activity.details.substr(0, this.activity.details.length - 7);
```
This is badly programmed. Would be fixed after https://github.com/nukeop/nuclear/pull/761 which doesn't hard code the length of `\nPaused`